### PR TITLE
fix #2 protocol aware caching

### LINKED ISSUE #2: Be protocol-agnostic
Although accessing a site over both HTTP and HTTPS isn't really recommended, the plugin currently doesn't work in this scenario ([see this support request](https://wordpress.org/support/topic/separate-caches-for-http-and-https?replies=2)).

The following changes need to be made:

* Take protocol into account in [`Cachify::_cache_hash`](https://github.com/pluginkollektiv/cachify/blob/master/inc/cachify.class.php#L1082) method.
* Take protocol into account in [ `Cachify_HDD::_file_path`](https://github.com/pluginkollektiv/cachify/blob/master/inc/cachify_hdd.class.php#L344) method
* Add protocol to `.htaccess` rewrite rules

### PR DESCRIPTION
**Pull request by stklcode**
A fixed version of PR #20 by @mcguffin.

* Prefix cannot be contained in regular URLs to avoid conflicts
* Protocol prefix has effect within every caching method

Fixes #2

### CODE REVIEW

**Comment by swissspidy** [comment]
Why is the prefix `s$-` and not something that makes more sense, like `https-`?

File: inc/cachify_hdd.class.php [hunk 1]
```diff
@@ -336,10 +336,13 @@ public static function _dir_size($dir = '.')
 
 	private static function _file_path($path = NULL)
 	{
+	  $prefix = is_ssl() ? 's$-' : '';
```
Comment (swissspidy) [inline inc/cachify_hdd.class.php]: Wrong indentation

**Comment by swissspidy** [comment]
Apart from the indentation and my question regarding the prefix this is 👍

**Comment by stklcode** [comment]
When I wrote this, i thought about the (even really unlikely) chance that one might be running on two domains like _example.com_ and _https-example.com_ with the same post url (e.g. non '/') which would result in the same hash for different pages. That's why I've chosen $ as a delimiter.

Issue may be related to multisite bug #23.

Changed it to a more speaking `https$` though (and fixed the indentation)

**Comment by swissspidy** [comment]
I don't know about the others, but I'm 100% sure there's nobody running `xy.com` _and_ `https-xy.com` _and_ running WordPress _and_ using Cachify. Therefore I prefer `https-` with a dash as a delimiter.

Would you mind changing that (in a new commit, so we don't lose history like before)? Thank you.

#23 is only somewhat related as #24 tries to fix #2 and #23. Left a comment.

**Comment by stklcode** [comment]
As I said, highly unlikely ... but not impossible. 

But I'm with you, the "risk" is acceptable, made a new commit.

**Pull request merged**